x-pack/plugins/maps/public/classes/layers/vector_layer/vector_layer.tsx

```diff
@@ -308,28 +372,37 @@ export class VectorLayer extends AbstractLayer {
     return await Promise.all(joinSyncs);
   }
 
-  _getSearchFilters(dataFilters, source, style) {
+  _getSearchFilters(
+    dataFilters: MapFilters,
+    source: IVectorSource,
+    style: IVectorStyle
+  ): VectorSourceRequestMeta {
     const fieldNames = [
       ...source.getFieldNames(),
       ...(style.getType() === LAYER_STYLE_TYPE.VECTOR ? style.getSourceFieldNames() : []),
       ...this.getValidJoins().map((join) => join.getLeftField().getName()),
     ];
 
+    const sourceQuery = this.getQuery() as MapQuery;
```

Comment:
would it make sense to change `ILayer#getQuery` to return `MapQuery | undefined` instead?

Now, this makes a double-hop:
- cast to `MapQuery` to make it work in the context of Maps
- null-check and conversion to undefined (since the case may fail)

Reply:
I went that route at first but decided to backtrack since it made the `getQuery` method non-symmetrical with other getters. For the most part, getters return an item or null. I think its a little weird for a getter to return undefined. I think we could take a look at all of our APIs and decide if we want a getters to return undefined but lets leave that up to a separate exercise.